ls1intum/Artemis
# [Bugfix] Fix client-side miscalculation of total score during the assessment of modeling submission

### Motivation and Context



We introduced `dropInfo` for the assessment object with this [PR](https://github.com/ls1intum/Apollon/pull/165). We only passed grading instruction id for this property. After dropping the grading instruction into UML element feedback, we only know `instructionId` in Artemis side. This cause  a problem to calculate the total score in the client side. Because, grading instructions have `usage count`. since we only passed the `grading instruction id`, the `usage count` returns `undefined` during the calculation (`calculateScoreForGradingInstructions` in `StructuredGradingCriterionService`). In Apollon side, my solution is setting the `dropInfo` with instruction itself.  Artemis can calculate the total score for all feedback elements properly.  

This PR aims to fix #3633. The issue is introduced on modeling assessment. I checked the other feedback elements (unreferenced feedback/text block feedback/inline-editor-feedback) using the grading instruction which has usage count. Total score is computed wrong only in modeling assessment. However, anyone can confirm that this is happening during the modeling submission assessment. 

PR will be updated after Apollon version update. => _**Updated**_

### Steps for Testing


For more details, you can check the screencast in the issue description.

1. Log in to Artemis
2. Navigate to Course Administration
3. Create/Find Modeling Exercise with Structured Grading Instructions
4. Define usage count for grading instruction.
5. Add an submission as a student.
6. As an instructor, go to Assessment Dashboard and Open the assessment.
7. Assess UML elements with grading instruction (you can drag & drop the instruction into the UML element)
8. Do not submit it, because server side calculation works, this is only for client side calculation. 
9. See that total score does not include the instruction credits anymore when it reaches the usage limit.

You can try this for other type of exercises(Text/File Upload/Programming with manual assessment). I did not encounter with this issue, and it needs to be confirmed.

### Test Coverage




nothings changed

### Screenshots
